app/services/chat.ts

```diff
@@ -0,0 +1,190 @@
+import { Service } from 'services/service';
+import { Inject } from 'util/injector';
+import { UserService } from 'services/user';
+import { getPlatformService } from 'services/platforms';
+import { CustomizationService, ICustomizationSettings } from 'services/customization';
+import electron from 'electron';
+import { YoutubeService } from 'services/platforms/youtube';
+import url from 'url';
+import { WindowsService } from 'services/windows';
+import { $t } from 'services/i18n';
+
+export class ChatService extends Service {
+  @Inject() userService: UserService;
+  @Inject() customizationService: CustomizationService;
+  @Inject() windowsService: WindowsService;
+
+  private chatView: Electron.BrowserView;
+
+  init() {
+    this.userService.userLogin.subscribe(() => this.initChat());
+    this.userService.userLogout.subscribe(() => this.deinitChat());
+
+    if (this.userService.isLoggedIn()) this.initChat();
+  }
+
+  refreshChat() {
+    this.navigateToChat();
+  }
+
+  mountChat(electronWindowId: number) {
+    if (!this.chatView) return;
+
+    const win = electron.remote.BrowserWindow.fromId(electronWindowId);
+
+    // This method was added in our fork
+    (win as any).addBrowserView(this.chatView);
+  }
+
+  setChatBounds(position: IVec2, size: IVec2) {
+    if (!this.chatView) return;
+
+    this.chatView.setBounds({
+      x: Math.round(position.x),
+      y: Math.round(position.y),
+      width: Math.round(size.x),
+      height: Math.round(size.y),
+    });
+  }
+
+  unmountChat(electronWindowId: number) {
+    if (!this.chatView) return;
+
+    const win = electron.remote.BrowserWindow.fromId(electronWindowId);
+
+    // This method was added in our fork
+    (win as any).removeBrowserView(this.chatView);
+  }
+
+  private initChat() {
+    if (this.chatView) return;
+
+    this.chatView = new electron.remote.BrowserView({
+      webPreferences: {
+        nodeIntegration: false,
+      },
+    });
+
+    this.navigateToChat();
+    this.bindWindowListener();
+    this.bindDomReadyListener();
+
+    this.customizationService.settingsChanged.subscribe(changed => {
+      this.handleSettingsChanged(changed);
+    });
+  }
+
+  private deinitChat() {
+    (this.chatView as any).destroy();
```

Comment:
I assume the cast is because it can be `null` (not initialized)? If so, can we conditionally call `destroy`?

Reply:
We're not using strict null checks, so that cast wouldn't be needed.  The reason for the cast is because electron's typings are incorrect for this API (not used by many people) so this method is not in the typings lol.